jax/_src/lax/linalg.py

```diff
@@ -888,7 +889,9 @@ def _lu_batching_rule(batched_args, batch_dims):
   x = batching.moveaxis(x, bd, 0)
   return lu_p.bind(x), (0, 0, 0)
 
-def _lu_cpu_gpu_translation_rule(getrf_impl, c, operand):
+
+def _lu_cpu_gpu_translation_rule(getrf_impl, pivots_to_permutation_impl, c,
```

Comment:
This is fine, but I think we can do even better. I suggest we make `lu_pivots_to_permutation` into a primitive, which allows it in turn to have per-backend lowering rules. You can then always use the `lower_fun` implementation in this translation rule.

The reason I suggest this is that some folks are calling `lu_pivots_to_permutation` explicitly:
https://cs.opensource.google/search?q=lu_pivots_to_permutation&sq=

and right now they would still get the slow implementation on GPU. Indeed the `scipy` API almost forces that upon us.

Happily we would not need, say, autodiff rules for such a primitive, just abstract eval, shape, and batching rules.